File: TestTask.UI/src/app/models/order.ts
import { Customer } from "./customer";
import { ProductInOrder } from "./productInOrder";
import { Status } from "./status";

export class Order{
    id = 0;
    date: Date = new Date;
    client: Customer = new Customer;
    status: Status = new Status;
    totalCost = 0;
    comment = "";
    productsInOrder: ProductInOrder[] = [];

    
}

File: TestTask/TestTask/Models/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace TestTask.Models;

public partial class Order
{
    [Key]
    public int OrderId { get; set; }

    public DateTime Date { get; set; }

    public int CustomerId { get; set; }

    public int StatusId { get; set; }

    public double TotalCost { get; set; }

    public string? Comment { get; set; }

    public virtual Customer? Customers { get; set; } = null!;

    public virtual ICollection<OrderProduct>? OrderProducts { get; set; } = new List<OrderProduct>();

    public virtual Status Statuses { get; set; } = null!;
}
